src/hotspot/share/classfile/javaClasses.cpp

```diff
@@ -2667,6 +2667,56 @@ void java_lang_Throwable::get_stack_trace_elements(Handle throwable,
   }
 }
 
+Handle java_lang_Throwable::get_cause_with_stack_trace(Handle throwable, TRAPS) {
+  // Call to JVM to fill in the stack trace and clear declaringClassObject to
+  // not keep classes alive in the stack trace.
+  // call this:  public StackTraceElement[] getStackTrace()
+  assert(throwable.not_null(), "shouldn't be");
+
+  JavaValue result(T_ARRAY);
+  JavaCalls::call_virtual(&result, throwable,
+                          vmClasses::Throwable_klass(),
+                          vmSymbols::getStackTrace_name(),
+                          vmSymbols::getStackTrace_signature(),
+                          CHECK_NH);
+  Handle stack_trace(THREAD, result.get_oop());
+  assert(stack_trace->is_objArray(), "Should be an array");
+
+  // If the original exception was in the bootstrap class loader, then use that
+  // otherwise use ExceptionInInitializerError as the cause with this stack trace,
+  // which might be a little wierd but still helpful.
+  Klass* tk = throwable->klass();
+  bool null_classloader = tk->class_loader() == nullptr;
+
+  Symbol* exception_name = null_classloader ?  tk->name() :
+                             vmSymbols::java_lang_ExceptionInInitializerError();
+
+  // Now create a the same exception with this stacktrace and thread name.
+  Symbol* message = java_lang_Throwable::detail_message(throwable());
+  ResourceMark rm(THREAD);
+  stringStream st;
+  if (!null_classloader) {
+    st.print("Exception %s%s ", tk->name()->as_klass_external_name(),
+             message == nullptr ? "" : ":");
+  }
+  if (message == NULL) {
+    st.print("[in thread \"%s\"]", THREAD->name());
+  } else {
+    st.print("%s [in thread \"%s\"]", message->as_C_string(), THREAD->name());
+  }
+
+  Handle h_cause = Exceptions::new_exception(THREAD, exception_name, st.as_string());
+
+  // If new exception returns a different exception while creating the exception, return null.
+  if (h_cause->klass()->name() != exception_name) {
```

Comment:
Some logging here to show that an exception occurred trying to do this would be useful I think.

Reply:
ok.  I should probably add back the table printing and logging.  I made it resizeable but I don't think I'll ever have to resize it so didn't put the code for that.